clickhouse/CMakeLists.txt

```diff
@@ -38,12 +38,18 @@ IF (WITH_OPENSSL)
     LIST(APPEND clickhouse-cpp-lib-src base/sslsocket.cpp)
 ENDIF ()
 
+IF (WITH_SYSTEM_CITYHASH AND NOT cityhash_FOUND)
+    SET (CITYHASH ${SYSTEM_CITYHASH})
```

Comment:
Perhaps we can add something similar to https://github.com/ClickHouse/clickhouse-cpp/blob/master/cmake/Findlz4.cmake to properly set a `cityhash::cityhash` imported library instead?

Reply:
That's actually a good idea. I pushed a `Findcityhash.cmake` and that should do it. I've verified this with a workflow in my fork: https://github.com/ZhongRuoyu/clickhouse-cpp/actions/runs/4840656304/jobs/8626442071#step:7:41. (Failure is due to a mismatched cityhash version, as discussed earlier -- but it's functional.)